torch/csrc/distributed/rpc/tensorpipe_agent.cpp

```diff
@@ -50,6 +57,50 @@ std::string guessAddress(tensorpipe::transport::uv::Context& uvContext) {
   return uvAddress;
 }
 
+namespace {
+
+std::unique_ptr<TransportRegistration> makeUvTransport() {
+  auto context = std::make_shared<tensorpipe::transport::uv::Context>();
+  std::string address = TensorPipeAgent::guessUvAddress(*context);
+  return std::make_unique<TransportRegistration>(
+      TransportRegistration{std::move(context), 1, std::move(address)});
+}
+
+C10_REGISTER_CREATOR(TensorPipeTransportRegistry, uv, makeUvTransport);
+
+#ifdef TP_ENABLE_SHM
+
+std::unique_ptr<TransportRegistration> makeShmTransport() {
+  auto context = std::make_shared<tensorpipe::transport::shm::Context>();
+  std::string address = TensorPipeAgent::createUniqueShmAddr();
+  return std::make_unique<TransportRegistration>(
+      TransportRegistration{std::move(context), 0, std::move(address)});
+}
+
+C10_REGISTER_CREATOR(TensorPipeTransportRegistry, shm, makeShmTransport);
+
+#endif
+
+std::unique_ptr<ChannelRegistration> makeBasicChannel() {
+  auto context = std::make_shared<tensorpipe::channel::basic::Context>();
+  return std::make_unique<ChannelRegistration>(
+      ChannelRegistration{std::move(context), 1});
+}
+
+C10_REGISTER_CREATOR(TensorPipeChannelRegistry, basic, makeBasicChannel);
+
+#ifdef TP_ENABLE_CMA
```

Comment:
Curious, what is a CMA channel?

Reply:
CMA stands for cross-memory attach, it's a funny syscall that Linux has (and, as far as I know, no other OS) which allows one process to read from/write to another process's private memory (provided the right permissions are in place). In a sense, this is pretty much what GDB does when you attach it to a running process.

Let me add a quick comment explaining what each transport/channel does.